### PR TITLE
fix(build-main): move ng dependencies to ROOT of Stark. Fix issue in build process. Rename tsconfig

### PR DESCRIPTION
**Pull request by SuperITMan**
Move ng dependencies and rxjs dep into the ROOT to avoid install of those shared in multiple
packages.
Rename tsconfig inside packages to tsconfig-build and remove root tsconfig.json file to
make IntelliJ working correctly

ISSUES CLOSED: #361, #362

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #361, #362

## What is the new behavior?
Fix build process + Avoid repetition of dependencies in the project.

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

**Comment by SuperITMan** [comment]
Apparently, there is an issue with tslint. I'll fix that now :)

File: packages/stark-core/package.json [hunk 1]
```diff
@@ -64,17 +56,18 @@
     "@angular/compiler-cli": "5.x",
     "@angular/core": "5.x",
     "@angular/platform-browser": "5.x",
-    "@angular/platform-browser-dynamic": "5.x"
+    "@angular/platform-browser-dynamic": "5.x",
+    "rxjs": "5.x"
```
Comment (christophercr) [inline packages/stark-core/package.json]: angular/router is also needed in Stark Core, could you please add it?

File: packages/stark-core/package.json [hunk 2]
```diff
@@ -64,17 +56,18 @@
     "@angular/compiler-cli": "5.x",
```
Comment (christophercr) [inline packages/stark-core/package.json]: Are we sure we need the angular/compiler-cli in Stark Core? and angular/compiler as well?

File: tsconfig.json [hunk 3]
```diff
@@ -1,11 +0,0 @@
-{
```
Comment (dsebastien) [inline tsconfig.json]: We should write some docs about the structure of our tsconfig files: #364

File: packages/stark-core/package.json (hunk 1 above)
Comment (SuperITMan) [reply packages/stark-core/package.json]: True, I forgot this. It's added now

File: packages/stark-core/package.json (hunk 2 above)
Comment (SuperITMan) [reply packages/stark-core/package.json]: I'm not sure it's needed. Currently, they are part of the stark-build and stark-core.

Comment (christophercr) [reply packages/stark-core/package.json]: To me, we should remove them unless they are really needed. So if you remove them and everything works just fine...

Comment (SuperITMan) [reply packages/stark-core/package.json]: For sure, because the deps will be available in the root :)
But ok, I'll remove those peerDeps from stark-core

**Pull request merged**